app/models/item.rb

```diff
@@ -9,20 +9,22 @@ class Item < ApplicationRecord
   has_many :invoices, through: :invoice_items
   has_many :transactions, through: :invoices
 
+  scope :find_match_by_name, ->(name) do
+    where('name Ilike ?', "%#{name}%")
+      .order('LOWER(name)')
+  end
+
```

Comment:
Used scope, I mostly prefer the controller to be as skinny as possible and have model logic in the model classes.